src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -421,6 +421,75 @@ public void updateFeedbackResponse(
                     "Trying to update a feedback response that does not exist.");
         }
 
+        copyOldResponseToNewResponse(newResponse, oldResponse);
+
+        if (!newResponse.recipientEmail.equals(oldResponse.recipientEmail) ||
+                !newResponse.giverEmail.equals(oldResponse.giverEmail)) {
+            recreateResponse(newResponse, oldResponse);
+        } else {
+            try {
+                frDb.updateFeedbackResponseOptimized(newResponse, oldResponseEntity);
+            } catch (EntityDoesNotExistException e) {
+                Assumption.fail();
+            }
+        }
+    }
+
+    /**
+     * Updates a {@link FeedbackResponse} using a {@link FeedbackResponseAttributes} <br>
+     * If the giver/recipient field is changed, the {@link FeedbackResponse} is
+     * updated by recreating the response<br>
+     * in order to prevent an id clash if the previous email is reused later on.
+     * @param updatedResponse 
+     * @param oldResponseEntity  a FeedbackResponse retrieved from the database 
+     * @throws EntityAlreadyExistsException  if trying to prevent an id clash by recreating a response,
+     *                                       a response with the same id already exist. 
+     * @throws InvalidParametersException               
+     */
+    public void updateFeedbackResponse(
+                        FeedbackResponseAttributes updatedResponse,
+                        FeedbackResponse oldResponseEntity)
+                                throws InvalidParametersException, EntityAlreadyExistsException {
+        if (oldResponseEntity == null) {
+            throw new NullPointerException("oldResponseEntity should not be null");
+        }
+        
+        // Create a copy.
+        FeedbackResponseAttributes newResponse = new FeedbackResponseAttributes(updatedResponse);
+        
+        FeedbackResponseAttributes oldResponse = new FeedbackResponseAttributes(oldResponseEntity);
+
+        copyOldResponseToNewResponse(newResponse, oldResponse);
```

Comment:
This `copyOldResponseToNewResponse` method can be error prone because it does not return anything but causes a side effect. It's not immediately clear which object is copied to which. Can we move it to `FeedbackResponseAttributes`? Or may be make it a constructor of `FeedbackResponseAttributes`?